### PR TITLE
Windows: Fix wildcard expansion after slash in filename

### PR DESCRIPTION
**Pull request by darstahl**
**\- What I did**

Fixes a bug in Wildcard expansion on Windows that caused wildcards at the beginning of nested directories (ie. immediately following a slash) to not be expanded for ADD and COPY instructions in a Dockerfile

**\- How I did it**

On Windows, `?` and`*` are invalid in file names, so don't need to be escaped.

**\- How to verify it**

Enabled a test that was failing as a result of this bug.

**\- Description for the changelog**

Fixed wildcard characters immediately following a path separator in Dockerfile ADD and COPY causing failures on a Windows daemon.

/cc @jhowardmsft @jstarks @duglin 

Signed-off-by: Darren Stahl darst@microsoft.com

### CODE REVIEW

**Comment by jhowardmsft** [comment]
LGTM. Esp as I helped debug the issue 😁

File: integration-cli/docker_cli_build_test.go [hunk 1]
```diff
@@ -863,10 +862,10 @@ func (s *DockerSuite) TestBuildCopyWildcard(c *check.C) {
 	ctx, err := fakeContext(fmt.Sprintf(`FROM busybox
 	COPY file*.txt /tmp/
 	RUN ls /tmp/file1.txt /tmp/file2.txt
-	RUN mkdir /tmp1
+	RUN sh -c "mkdir /tmp1"
```
Comment (vdemeester) [inline integration-cli/docker_cli_build_test.go]: why this change ? 👼

**Comment by vdemeester** [comment]
One question, otherwise LGTM 🐸
/cc @duglin

File: integration-cli/docker_cli_build_test.go (hunk 1 above)
Comment (duglin) [inline integration-cli/docker_cli_build_test.go]: an a few unrelated questions... does the busybox for Windows have "sh" ?  And does the busybox for windows support / and \  as dir separators ?

**Comment by duglin** [comment]
LGTM aside from @vdemeester's question on the testcase

File: integration-cli/docker_cli_build_test.go (hunk 1 above)
Comment (jhowardmsft) [inline integration-cli/docker_cli_build_test.go]: It's because windows launches non-JSON commands in a dockerfile with cmd /s, so would attempt to use the mkdir from cmd. This forces it into the busybox mkdir. And yes, Windows busybox does have sh. But I don't believe it supports \, can check when in front of a computer later. Alternate solution is to use JSON syntax - six of one, half a dozen of the other...

Comment (vdemeester) [inline integration-cli/docker_cli_build_test.go]: Hum, I would think that using the JSON syntax would make more sence then (it's more clear than using `sh -c` in this case).

Comment (darrenstahlmsft) [inline integration-cli/docker_cli_build_test.go]: Update to use JSON syntax.

**Comment by jhowardmsft** [comment]
Still LGTM

**Comment by duglin** [comment]
still LGTM but I'll defer to @vdemeester for the merge

**Review by vdemeester — APPROVED** [review]

**Comment by vdemeester** [comment]
LGTM :frog:

**Comment by jhowardmsft** [comment]
All green. Merging.

File: integration-cli/docker_cli_build_test.go (hunk 1 above)
Comment (darrenstahlmsft) [inline integration-cli/docker_cli_build_test.go]: @duglin In response to your question about busybox and using `\`. It works as a path separator, but is explicitly called out as not valid, so the answer is no, but it works anyway.
